k4FWCore/components/UniqueIDGenSvc.h

```diff
@@ -41,10 +42,17 @@ class UniqueIDGenSvc : public extends<Service, IUniqueIDGenSvc> {
   size_t getUniqueID(const event_num_t evt_num, const run_num_t run_num, const std::string& name) const override;
 
 private:
-  Gaudi::Property<seed_t>                           m_seed{this, "Seed", {123456789}};
-  mutable std::unordered_set<size_t, std::identity> m_uniqueIDs;
-  mutable std::mutex                                m_mutex;
-  Gaudi::Property<bool>                             m_throwIfDuplicate{this, "ThrowIfDuplicate", {true}};
+  Gaudi::Property<seed_t> m_seed{this, "Seed", {123456789}};
+  mutable std::unordered_map<size_t, std::tuple<event_num_t, run_num_t, std::string>, std::identity> m_uniqueIDs;
+  mutable std::mutex                                                                                 m_mutex;
+  Gaudi::Property<bool>                                                                              m_checkDuplicates{
+      this, "CheckDuplicates", m_isDebugBuild,
+      "Caches obtained ID and throws an exception if a duplicate would be returned"};
+#ifdef NDEBUG
+  constexpr static bool m_isDebugBuild = false;
+#else
+  constexpr static bool m_isDebugBuild = true;
+#endif
```

Comment:
```suggestion
  Gaudi::Property<bool>                                                                              m_checkDuplicates{
      this,
      "CheckDuplicates",
      // Default value for release and debug builds
#ifdef NDEBUG
      false;
#else
      true;
#endif
      "Caches obtained ID and throws an exception if a duplicate would be returned"};
```
Isn't this simpler without the extra `m_isDebugBuild`?

Reply:
I don't think there is that much of a difference but there is a bit less of indirection with this approach. Can you give it a try and we see it on the diff?

Reply:
Here it is